clients/metadata/device_profile.go

```diff
@@ -126,7 +126,7 @@ func (dpc *DeviceProfileRestClient) Update(dp models.DeviceProfile, ctx context.
 }
 
 func (dpc *DeviceProfileRestClient) Upload(yamlString string, ctx context.Context) (string, error) {
-	return clients.PostRequest(dpc.url+"/upload", []byte(yamlString), clients.ContentYaml, ctx)
+	return clients.PostRequest(dpc.url+"/upload", []byte(yamlString), ctx)
```

Comment:
Notice that the previous implementation passed in the YAML content type. We've lost that with this change. Suggest you check the `Context` for `Content-Type`. If it's not there, then add it with the appropriate YAML value. The subsequent handling inside of `PostRequest` should then pick it up.

Reply:
`fromContext` is still out of scope here, but it shouldn't be a problem to always set the `Content-Type` to YAML, right? We are uploading YAML right here, aren't we?